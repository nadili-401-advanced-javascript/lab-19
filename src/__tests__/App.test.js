import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App';
import Converter from '../components/Converter';
import { configure, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

// uses react-test-renderer for snap-shot test
describe('App component', () => {
    it('renders correctly', () => {
        const page = renderer.create(<App />).toJSON();
        expect(page).toMatchSnapshot();
    });
});

// Enzyme full mount test
it('has a Converter', () => {
    let component = mount(<App />);
    expect(component.containsMatchingElement(
        <span> is equal to </span>
    )).toBeTruthy()
});

// uses render function from Enzyme
describe('Converter component', () => {
    it('correctly renders converter', () => {
        let component = render(<Converter />);
        expect(component.find('h4.converter-heading')).toBeDefined();
    });
});

// 
// it('adds one to Counter', () => {
//     let component = mount(<App />);
//     let input1 = component.find('label input#x');
//     let input2 = component.find('label input#y');
//     addBtn.simulate('click');
//     expect(component.find('div.bordered-box h2').text()).toBe(
//         'Your current count is: 1',
//     );
// });

