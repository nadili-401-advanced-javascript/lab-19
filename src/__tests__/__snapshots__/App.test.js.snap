// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    Multi-Purpose Converter
  </h1>
  <div
    className="converter"
  >
    <h4
      className="converter-heading"
    >
      Converting 
      Miles
       to/from 
      Kilometers
    </h4>
    <label
      className="x-label"
    >
      <input
        id="x"
        onChange={[Function]}
        step="any"
        type="number"
        value=""
      />
      Miles
    </label>
    <span
      className="mid-label"
    >
       is equal to 
    </span>
    <label
      className="y-label"
    >
      <input
        id="y"
        onChange={[Function]}
        step="any"
        type="number"
        value=""
      />
      Kilometers
    </label>
  </div>
  <div
    className="converter"
  >
    <h4
      className="converter-heading"
    >
      Converting 
      Cups
       to/from 
      Fluid Ounces
    </h4>
    <label
      className="x-label"
    >
      <input
        id="x"
        onChange={[Function]}
        step="any"
        type="number"
        value=""
      />
      Cups
    </label>
    <span
      className="mid-label"
    >
       is equal to 
    </span>
    <label
      className="y-label"
    >
      <input
        id="y"
        onChange={[Function]}
        step="any"
        type="number"
        value=""
      />
      Fluid Ounces
    </label>
  </div>
  <div
    className="converter"
  >
    <h4
      className="converter-heading"
    >
      Converting 
      Inches
       to/from 
      Centimeters
    </h4>
    <label
      className="x-label"
    >
      <input
        id="x"
        onChange={[Function]}
        step="any"
        type="number"
        value=""
      />
      Inches
    </label>
    <span
      className="mid-label"
    >
       is equal to 
    </span>
    <label
      className="y-label"
    >
      <input
        id="y"
        onChange={[Function]}
        step="any"
        type="number"
        value=""
      />
      Centimeters
    </label>
  </div>
</div>
`;
